com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/managers/BazelBuildSupport.java

```diff
@@ -0,0 +1,156 @@
+        try {
+            updateInternal(project, force, monitor);
+
+        } catch (CoreException ex) {
+            throw ex;
+
+        } catch (AssertionFailedException ex) {
+            // Bazel can't work with the provided project.
+            // Just skip it.
+
+        } catch (NoSuchElementException ex) {
+            // No bazel importers found. This is definitely illegal situation
+            // which is not clear how to solve right now.
+            // Just skip it.
+        }
+
+    }
+
+    protected void updateInternal(IProject project, boolean force, IProgressMonitor monitor) throws CoreException {
+
+        Assert.isTrue(applies(project));
+
+        final IProjectImporter importer = obtainBazelImporter().get();
+
+        //TODO is false becasue rootFolder is NULL
+        Assert.isTrue(importer.applies(monitor));
+
+        importer.importToWorkspace(monitor);
+    }
+
+    @Override
+    public boolean isBuildFile(IResource resource) {
+        return resource != null && resource.getProject() != null && resource.getType() == IResource.FILE
+                && (resource.getName().endsWith(BAZEL_PROJECT_FILE_SUFIX) || resource.getName().equals(BUILD_FILE)
+                        || resource.getName().equals(BUILD_FILE));
+    }
+
+    @Override
+    public boolean isBuildLikeFileName(String fileName) {
+        return fileName.endsWith(BAZEL_PROJECT_FILE_SUFIX) || fileName.equals(BUILD_FILE)
+                || fileName.equals(BUILD_FILE);
+    }
+
+    @Override
+    public List<String> getWatchPatterns() {
+        return WATCH_FILE_PATTERNS;
+    }
+
+    @Override
+    public boolean fileChanged(IResource resource, CHANGE_TYPE changeType, IProgressMonitor monitor)
+            throws CoreException {
+        if (resource == null || !applies(resource.getProject())) {
+            return false;
+        }
+        return IBuildSupport.super.fileChanged(resource, changeType, monitor) || isBuildFile(resource);
+    }
+
+    ///////
+
+    private Optional<IProjectImporter> obtainBazelImporter() {
+        return ExtensionsExtractor.<IProjectImporter>extractOrderedExtensions(IConstants.PLUGIN_ID, "importers")
+                .stream().filter(importer -> importer instanceof BazelProjectImporter).findFirst();
+    }
+
+    @Override
+    public String buildToolName() {
+        return "Bazel";
+    }
+
+    @Override
+    public boolean hasSpecificDeleteProjectLogic() {
+        return true;
+    }
+
+    @Override
+    public void deleteInvalidProjects(Collection<IPath> rootPaths, ArrayList<IProject> deleteProjectCandates,
+            IProgressMonitor monitor) {
+        for (IProject project : deleteProjectCandates) {
+            if (applies(project)) {
+                IPath projectLocation = getProjectLocation(project);
+
+                if (ResourceUtils.isContainedIn(projectLocation, rootPaths)) {
+                    // NOP - if the project is contained in the root path, it's a valid project
```

Comment:
We don't need this comment which simply duplicates what we have in the line below